File: Two_Pointers/Valid-Palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        s = s.replaceAll("[^a-zA-Z0-9]", "");
        s = s.toLowerCase();
        char[] cs = s.toCharArray();

        int l = 0;
        int r = cs.length - 1;

        if (s.length() == 1) { return true; }

        while (l <= r) {
            if (cs[l] == cs[r]) {
                l++;
                r--;
            } else {
                return false;
            }
        } 

        return (l >= r);

    }
}

File: Two_Pointers/Valid-Palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string aux;
        for (char c : s) {
            if (isalnum(c)) { aux += tolower(c); }
        }

        int r = aux.length() - 1;

        for (int l = 0; l <= r; l++) {
            if (tolower(aux[l]) != tolower(aux[r])) { return false; }
            r--;
        }
        
        return true;

    }
};
